endpoints/test/openapi_generator_test.py

```diff
@@ -1360,8 +1468,203 @@ def noop_get(self, unused_request):
+                            "auth0-67eeef35": []
+                        }
+                    ],
+                }
+            },
+        },
+        'definitions': {
+            ALL_FIELDS: {
+                'type': 'object',
+                'properties': {
+                    'bool_value': {
+                        'type': 'boolean',
+                    },
+                    'bytes_value': {
+                        'type': 'string',
+                        'format': 'byte',
+                    },
+                    'datetime_value': {
+                        'type': 'string',
+                        'format': 'date-time',
+                    },
+                    'double_value': {
+                        'type': 'number',
+                        'format': 'double',
+                    },
+                    'enum_value': {
+                        'type': 'string',
+                        'enum': [
+                            'VAL1',
+                            'VAL2',
+                        ],
+                    },
+                    'float_value': {
+                        'type': 'number',
+                        'format': 'float',
+                    },
+                    'int32_value': {
+                        'type': 'integer',
+                        'format': 'int32',
+                    },
+                    'int64_value': {
+                        'type': 'string',
+                        'format': 'int64',
+                    },
+                    'message_field_value': {
+                        '$ref': self._def_path(NESTED),
+                        'description':
+                            'Message class to be used in a message field.',
+                    },
+                    'sint32_value': {
+                        'type': 'integer',
+                        'format': 'int32',
+                    },
+                    'sint64_value': {
+                        'type': 'string',
+                        'format': 'int64',
+                    },
+                    'string_value': {
+                        'type': 'string',
+                    },
+                    'uint32_value': {
+                        'type': 'integer',
+                        'format': 'uint32',
+                    },
+                    'uint64_value': {
+                        'type': 'string',
+                        'format': 'uint64',
+                    },
+                },
+            },
+            "OpenApiGeneratorTestNested": {
+                "type": "object",
+                "properties": {
+                    "int_value": {
+                        "format": "int64",
+                        "type": "string"
+                    },
+                    "string_value": {
+                        "type": "string"
+                    },
+                },
+            },
+        },
+        "securityDefinitions": {
+            "auth0": {
```

Comment:
Is this definition necessary?

Reply:
No, but it does no harm.

The openapi generator directly mutates the ending dictionary structure instead of using a Builder pattern like the Java framework does. This means we need the un-modified security definition in the structure until we're done processing methods.

At that point it would certainly be possible to remove unused security definitions, but I felt that was an unnecessary optimization. If you disagree, I can add it.